src/migmose/mig/reducednestednachrichtenstruktur.py

```diff
@@ -2,16 +2,135 @@
 contains class for trees consisting of segments of mig tables
 """
 
-from typing import Optional
+import json
+from pathlib import Path
+from typing import Any, Optional
 
+from loguru import logger
+from maus.edifact import EdifactFormat
 from pydantic import BaseModel
 
 from migmose.mig.nachrichtenstrukturzeile import NachrichtenstrukturZeile
+from migmose.mig.nestednachrichtenstruktur import NestedNachrichtenstruktur
 
 
 class ReducedNestedNachrichtenstruktur(BaseModel):
     """will contain the tree structure of nachrichtenstruktur tables"""
 
     header_linie: Optional[NachrichtenstrukturZeile] = None
-    segmente: list[NachrichtenstrukturZeile]
-    segmentgruppe: list["ReducedNestedNachrichtenstruktur"]
+    segmente: list[Optional[NachrichtenstrukturZeile]] = []
+    segmentgruppen: list[Optional["ReducedNestedNachrichtenstruktur"]] = []
```

Comment:
in the lines below you use the syntax ` int | None`.
I would prefer to stick to one notation: Either `list[Optional[int]]` or `list[int | None]`

Reply:
https://gitkraken.dev/link/Z2l0a3Jha2VuOi8vcmVwb2xpbmsvMjI3MWY3NTc2NWQyNjgzZDhhNTE4MjA1YTgwZjZhMjBhZWI0M2I2MS9jb21taXQvYWU1OWRjNThlM2RhMWFmYjM2ZjFkZWJkZDZmZmExYzExMzFkOTVlYz91cmw9aHR0cHMlM0ElMkYlMkZnaXRodWIuY29tJTJGSG9jaGZyZXF1ZW56JTJGbWlnbW9zZS5naXQ%3D?origin=gitkraken

Reply:
https://github.com/Hochfrequenz/migmose/pull/41/commits/ae59dc58e3da1afb36f1debdd6ffa1c1131d95ec